base/src/TestSignalGeneratorSrc.cpp

```diff
@@ -97,14 +110,14 @@ void TestSignalGenerator::setMetadata(framemetadata_sp &metadata)
     }
 }
 
-TestSignalGeneratorProps TestSignalGenerator::getProps()
+void TestSignalGenerator::setProps(TestSignalGeneratorProps &_props)
 {
```

Comment:
this is wrong - use addPropsToQueue and handlePropsChange. we must change the props after the existing frames in queue are processed

Reply:
I have added addPropsToQueue inside the setProps and also created handlePropsChange function.